service/src/integrationTest/kotlin/fi/espoo/evaka/messaging/MessageIntegrationTest.kt

```diff
@@ -1004,6 +1004,110 @@ class MessageIntegrationTest : FullApplicationTest(resetDbBeforeEach = true) {
         assertEquals(2, unreadMessagesCount(person2))
     }
 
+    @Test
+    fun `unread message count excludes messages beyond employee access limit`() {
+        val area = DevCareArea(shortName = "testArea")
+        val unit =
+            DevDaycare(areaId = area.id, enabledPilotFeatures = setOf(PilotFeature.MESSAGING))
+        val group = DevDaycareGroup(daycareId = unit.id, startDate = placementStart.minusYears(1))
+        val employee = DevEmployee()
+        val employeeUser =
+            AuthenticatedUser.Employee(id = employee.id, roles = setOf(UserRole.STAFF))
+        val admin = DevEmployee(roles = setOf(UserRole.ADMIN))
+        val adminUser = AuthenticatedUser.Employee(id = admin.id, roles = setOf(UserRole.ADMIN))
+        val daycareGroupAcl =
+            DevDaycareGroupAcl(
+                groupId = group.id,
+                employeeId = employee.id,
+                created = HelsinkiDateTime.of(LocalDate.of(2022, 5, 14), LocalTime.of(12, 0)),
+            )
+        val adult = DevPerson()
+        val adultUser = AuthenticatedUser.Citizen(adult.id, CitizenAuthLevel.STRONG)
+        val child = DevPerson()
+        db.transaction { tx ->
+            tx.insert(area)
+            tx.insert(unit)
+            tx.insert(group)
+            tx.insert(employee)
+            tx.insert(admin)
+            tx.insert(daycareGroupAcl)
+            tx.insert(adult, DevPersonType.ADULT)
+            tx.insert(child, DevPersonType.CHILD)
+            tx.insert(DevGuardian(adult.id, child.id))
+            val placementId =
+                tx.insert(
+                    DevPlacement(
+                        childId = child.id,
+                        unitId = unit.id,
+                        startDate = placementStart.minusMonths(5),
+                        endDate = placementEnd,
+                    )
+                )
+            tx.insert(
+                DevDaycareGroupPlacement(
+                    daycarePlacementId = placementId,
+                    daycareGroupId = group.id,
+                    startDate = placementStart.minusMonths(5),
+                    endDate = placementEnd,
+                )
+            )
+        }
+
+        val groupAccount = db.transaction { tx -> tx.createDaycareGroupMessageAccount(group.id) }
+        val municipalAccount = db.transaction { tx -> tx.createMunicipalMessageAccount() }
+
+        // Message thread beyond employee access limit (1 week before daycare group acl creation)
+        postNewThread(
+            title = "Juhannus",
+            message = "Juhannus tulee pian",
+            messageType = MessageType.MESSAGE,
+            sender = groupAccount,
+            recipients = listOf(MessageRecipient.Child(child.id)),
+            user = employeeUser,
+            now = HelsinkiDateTime.of(LocalDate.of(2022, 5, 6), LocalTime.of(12, 11)),
```

Comment:
make the access limit week a variable and use it here accessStartsDate.minusDays(7) or similar